packages/cli/src/lib/runPreBuildTasks.js

```diff
@@ -0,0 +1,32 @@
+import path from 'path'
+
+import { getProject } from '@redwoodjs/structure'
+
+const project = getProject()
+
+function generateTypeDef(filename, contents) {
+  const { host } = project
+  host.writeFileSync(path.join(host.paths.types, filename), contents)
+}
+
+function generateCurrentUserTypes() {
+  generateTypeDef(
+    'currentUser.d.ts',
+    project.host.readFileSync(
+      path.join(__dirname, './templates/currentUser.d.ts.template')
+    )
+  )
+}
+
+/**
+ * @description This function houses all the tasks that
+ * need to run before rw build/rw dev
+ */
+
+function runPreBuildTasks() {
+  if (project.isTypeScriptProject) {
```

Comment:
Hmpf, it would've been nice if there was a way for JavaScript users to manually type their currentUser, but maybe that's a stretch goal.

Reply:
Maybe we can remove this flag and use the magic type import comments
```
 /** @type {import('@redwoodjs/auth').CurrentUser} */
```